odoo/addons/base/models/ir_actions_report.py

```diff
@@ -198,6 +198,44 @@ def retrieve_attachment(self, record):
                 ('res_id', '=', record.id)
         ], limit=1)
 
+    def _prepare_pdf_report_attachment_vals_list(self, report, streams):
```

Comment:
maybe keep this hook method closer to the upper method.
This is line 200ish and `_render_qweb_pdf` is on line 800ish :)